component/kafka/kafka_test.go

```diff
@@ -88,3 +89,35 @@ func Test_DefaultConsumerSaramaConfig(t *testing.T) {
 	require.NoError(t, err)
 	require.NotEqual(t, sarama.ReadCommitted, sc.Consumer.IsolationLevel)
 }
+
+func TestDeduplicateMessages(t *testing.T) {
+	message := func(key, val string) Message {
+		return NewMessage(
+			context.TODO(),
```

Comment:
`context.TODO()` is used when it's unclear which Context to use. I think here we should rather use `context.Background()`.